crates/ruff_python_ast/src/nodes.rs

```diff
@@ -1484,26 +1484,10 @@ impl fmt::Debug for FStringFlags {
 #[derive(Clone, Debug, PartialEq)]
 pub struct FString {
     pub range: TextRange,
-    pub elements: Vec<FStringElement>,
+    pub elements: FStringElements,
     pub flags: FStringFlags,
 }
 
-impl FString {
-    /// Returns an iterator over all the [`FStringLiteralElement`] nodes contained in this f-string.
-    pub fn literals(&self) -> impl Iterator<Item = &FStringLiteralElement> {
-        self.elements
-            .iter()
-            .filter_map(|element| element.as_literal())
-    }
-
-    /// Returns an iterator over all the [`FStringExpressionElement`] nodes contained in this f-string.
-    pub fn expressions(&self) -> impl Iterator<Item = &FStringExpressionElement> {
-        self.elements
-            .iter()
-            .filter_map(|element| element.as_expression())
-    }
-}
```

Comment:
I would be tempted to keep these methods as simple wrappers, e.g.

```rs
impl FString {
    pub fn literals(&self) -> impl Iterator<Item = &FStringLiteralElement> {
        self.elements.literals()
    }

    pub fn expressions(&self) -> impl Iterator<Item = &FStringExpressionElement> {
        self.elements.expressions()
    }
}
```

As I think `for literal in f_string.literals()` feels like a slightly more natural API for downstream users than `for literal in f_string.elements.literals()`. But this is just a nit.

Reply:
Yeah, I thought about that but then I'm not sure if there's any more benefit to implement the newtype. Also, the reason I think `f_string.elements.literals()` API is better is because `elements` is a public field, so the user can either get all of the element or apply a filter on top of it (`elements.literals()`). And, this wrapper would need to be implemented for `FStringFormatSpec` as well.

Reply:
Maybe part of the issue is the name of the method... what if `FStringElements.literals()` was renamed to `iter_literal_elements()`? And `expressions` renamed to `iter_expression_elements`? That might make it slightly clearer -- I think `for literal in f_string.elements.iter_literal_elements()` would read more naturally to me, although it's obviously slightly more verbose

But, as I say, this is just a nit :-)

Reply:
That suggestion is valid although I'm fine with this and this naming pattern has been used in a lot of similar places ;)

Thanks for the suggestion!